src/transformers/models/pop2piano/tokenization_pop2piano.py

```diff
@@ -0,0 +1,428 @@
+# coding=utf-8
+# Copyright 2023 The Pop2Piano Authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization class for Pop2Piano."""
+
+import json
+import os
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import pretty_midi
+import torch
+
+from ...feature_extraction_utils import BatchFeature
+from ...tokenization_utils import AddedToken, PreTrainedTokenizer
+from ...utils import logging, to_numpy
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {
+    "vocab_file": "vocab.json",
+}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "sweetcocoa/pop2piano": "https://huggingface.co/sweetcocoa/pop2piano/blob/main/vocab.json",
+    },
+}
+
+
+class Pop2PianoTokenizer(PreTrainedTokenizer):
+    """
+    Constructs a Pop2Piano tokenizer. This tokenizer does not require training.
+
+    This tokenizer inherits from [`PreTrainedTokenizer`] which contains most of the main methods. Users should refer
+    to: this superclass for more information regarding those methods. However the code does not allow that and only
+    supports composing from various genres.
+
+    Args:
+        vocab_file (`str`):
+            Path to the tokenizer file which contains token-ids such as `TOKEN_SPECIAL`, `DEFAULT_VELOCITY`.
+        vocab_size_special (`int`, *optional*, defaults to 4):
+            Number of special values.
+        vocab_size_note (`int`, *optional*, defaults to 128):
+            Number of MIDI note tokens. Only the 88 pitches corresponding to piano keys are actually used.
+        vocab_size_velocity (`int`, *optional*, defaults to 2):
+            Number of velocity tokens.
+        vocab_size_time (`int`, *optional*, defaults to 100):
+            Number of beat shift tokens. Beat shifts indicate the relative time shift within the segment quantized into
+            8th-notes (half-beats).
+        num_bars (`int`, *optional*, defaults to 2):
+            Determines cutoff_time_idx in for each token.
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+
+    def __init__(
+        self,
+        vocab_file,
+        vocab_size_special: int = 4,
+        vocab_size_note: int = 128,
+        vocab_size_velocity: int = 2,
+        vocab_size_time: int = 100,
```

Comment:
As I believe that since we need to change the file at checkpoint, I have opened a [PR](https://huggingface.co/sweetcocoa/pop2piano/discussions/3)(on checkpoint repo) regarding this, let us wait until the PR is merged then I will update the code. Please ignore this until that. 
cc : @sweetcocoa